client/src/components/Profile/__test__/ProfileSettingsModal.test.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import ProfileSettingsModal from '../ProfileSettingsModal';
+
+describe('ProfileSettingsModal', () => {
+  it('should render correctly', () => {
+    const output = render(
+      <ProfileSettingsModal
+        isSettingsVisible={true}
+        content={<div>content</div>}
+        settingsTitle="Settings Title"
+        onSave={jest.fn()}
+        onCancel={jest.fn()}
+      />,
+    );
+
+    expect(output.getByTestId('profile-settings-modal')).toMatchSnapshot();
```

Comment:
Are you sure that you can't reach this element without testId? By role, some content, etc.

Reply:
Apparently I can use `dialog` role, it wont include few wrapper divs, but I think it doesn't matter.